File: oauth-service/src/main/java/a/gleb/oauthservice/db/entity/Account.java
package a.gleb.oauthservice.db.entity;

import lombok.Getter;
import lombok.RequiredArgsConstructor;
import lombok.Setter;
import lombok.ToString;

import javax.persistence.*;
import java.util.UUID;

import static javax.persistence.GenerationType.AUTO;

@Entity(name = "accounts")
@Getter
@Setter
@ToString
@RequiredArgsConstructor
public class Account {

    @Id
    @GeneratedValue(strategy = AUTO)
    @Column(name = "id", nullable = false)
    private UUID id;

    @Column(name = "username")
    private String username;

    @Column(name = "password")
    private String password;

    @Column(name = "first_name")
    private String firstName;

    @Column(name = "last_name")
    private String lastName;

    @Column(name = "role")
    @Enumerated(EnumType.STRING)
    private AccountRoles role;

    @Column(name = "email")
    private String email;

    @Column(name = "enabled")
    private boolean enabled = false;

}

File: oauth2-manager/src/main/kotlin/a/gleb/oauth2manager/db/entity/Account.kt
package a.gleb.oauth2manager.db.entity

import java.util.*
import javax.persistence.Column
import javax.persistence.Entity
import javax.persistence.GeneratedValue
import javax.persistence.GenerationType.AUTO
import javax.persistence.Id

@Entity(name = "accounts")
data class Account(

    @Id
    @GeneratedValue(strategy = AUTO)
    @Column(name = "id", nullable = false)
    var id: UUID,
    @Column(name = "username")
    var username: String,
    @Column(name = "password")
    var password: String,
    @Column(name = "first_name")
    var firstName: String? = null,
    @Column(name = "last_name")
    var lastName: String? = null,
    @Column(name = "role")
    var roles: String,
    @Column(name = "email")
    var email: String,
    @Column(name = "enabled")
    var enabled: Boolean = false
)
